log/sentry.go

```diff
@@ -88,3 +89,45 @@ func (c closer) Close() error {
 		ErrSentryFlushFailed,
 	)
 }
+
+// ErrorWithSentry logs a message with some additional context,
+// then sends the error to Sentry.
+//
+// The variadic key-value pairs are treated as they are in With.
+// and will also be sent to sentry.
+// Note that zap.Field is not supported here and they must be in pairs.
+//
+// If a sentry hub is attached to the context object passed in
+// (it will be if the context object is from baseplate hooked request context),
+// that hub will be used to do the reporting.
+// Otherwise the global sentry hub will be used instead.
+func ErrorWithSentry(ctx context.Context, msg string, err error, keysAndValues ...interface{}) {
+	hub := sentry.GetHubFromContext(ctx)
+	if hub == nil {
+		hub = sentry.CurrentHub()
+	}
+	if len(keysAndValues) > 0 {
+		if len(keysAndValues)%2 == 1 {
+			// There's a dangling key, log and append an empty value to avoid panic.
+			Errorw(
+				"Dangling key in ErrorWithSentry",
+				"keysAndValues", keysAndValues,
+			)
+			keysAndValues = append(keysAndValues, "")
+		}
+		hub = hub.Clone()
+		hub.ConfigureScope(func(scope *sentry.Scope) {
+			for i := 0; i < len(keysAndValues); i++ {
```

Comment:
Shouldn't this be `i += 2`?

```suggestion
			for i := 0; i < len(keysAndValues); i+=2 {
```

Reply:
Yes 🤦

Reply:
Easy mistake to make 😅